server/src/main/java/org/elasticsearch/action/admin/indices/refresh/RefreshResponse.java

```diff
@@ -21,18 +21,36 @@
 
 import org.elasticsearch.action.support.DefaultShardOperationFailedException;
 import org.elasticsearch.action.support.broadcast.BroadcastResponse;
+import org.elasticsearch.common.xcontent.ConstructingObjectParser;
+import org.elasticsearch.common.xcontent.XContentParser;
 
+import java.util.Arrays;
 import java.util.List;
 
 /**
  * The response of a refresh action.
  */
 public class RefreshResponse extends BroadcastResponse {
 
+    private static final ConstructingObjectParser<RefreshResponse, Void> PARSER = new ConstructingObjectParser<>("refresh", true,
+        arg -> {
+            BroadcastResponse response = (BroadcastResponse) arg[0];
```

Comment:
I wonder if there's a way to avoid creating an instance of `BroadcastResponse` given that we use it only to transport the parsed values, maybe we can get multiple arguments directly.

Reply:
I tried it out and understood that it is not possible. It is fine to do it the way you are doing it, by using an intermediate broadcast response object.